piqe_ocp_lib/api/tasks/operator_ops.py

```diff
@@ -15,68 +17,93 @@
         """
-        A helper method that creates a placeholder project that will contain the
-        subscription and operator group objects necessary to install an operator
-        For openshift 4.1, the operator installation only succeeds if these objects
-        are installed in the opeshift-marketplace namespace. In openshift 4.2+, we
-        create a project with the following naming convention:
-        test + operator name + install mode + og-sub-project
+        A helper method that creates the namespace where both the subscription and operator
+        group objects will be created to install an operator.
         """
-        og_name = f"{operator_name}-{install_mode.lower()}-og"
-        og_namespace = f"test-{operator_name}-{install_mode.lower()}-og-sub-project"
-        logger.info(f"Creating Project: {og_namespace} that will hold the subscription and operator group")
-        assert self.proj_obj.create_a_project(og_namespace)
-        assert self.og_obj.create_operator_group(og_name, og_namespace, target_namespaces)
-        return og_name, og_namespace
-
-    def add_operator_to_cluster(self, operator_name, target_namespaces=[]):
+        if not operator_namespace:
+            operator_namespace = self.ohp_obj.get_channel_suggested_namespace(operator_name, channel_name)
+        if not operator_namespace:
+            operator_namespace = f"openshift-{operator_name}"
+        assert self.proj_obj.create_a_namespace(operator_namespace)
+        return operator_namespace
+
+    def _create_og(self, operator_name: str,
+                   channel_name: str,
+                   operator_namespace: str,
+                   target_namespaces: Union[list, str]) -> Tuple[str, str]:
+        """
+        A helper method that creates the operator group in the generated operator namespace
         """
-        Install an operator in a list of targeted namespaces
+        derived_install_mode = self._derive_install_mode_from_target_namespaces(target_namespaces)
+        if not self.ohp_obj.is_install_mode_supported_by_channel(operator_name, channel_name, derived_install_mode):
+            err_msg = f"The specified channel doesn't support {channel_name} installs"
+            logger.exception(err_msg)
+            raise UnsupportedInstallMode(err_msg)
+        else:
+            operator_namespace = self._create_operator_namespace(operator_name, operator_namespace, channel_name)
+            og_name = f"{operator_name}-og"
+            assert self.og_obj.create_operator_group(og_name, operator_namespace, target_namespaces)
+        return og_name, operator_namespace
+
+    def add_operator_to_cluster(self, operator_name: str,
+                                channel_name: str = '',
+                                operator_namespace: str = '',
+                                target_namespaces: Union[list, str] = []) -> bool:
```

Comment:
```suggestion
                                target_namespaces: Union[List[str], str] = []) -> bool:
```
sadly using the native container classes (`dict`, `list`, etc.) aren't supported in <3.9 https://mypy.readthedocs.io/en/stable/builtin_types.html#generic-types